### PR TITLE
Add an implementation, `network::devnet::DevNet`, for trait `AuthorizedNetwork`.

### PR DESCRIPTION
**Pull request by lemonholic**
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix
- [ ] Dependency, Environment, Build related update 
- [ ] Documentation
- [x] Other tiny fixes

## Describe your changes
- New struct: `simperby_network::devnet::DevNet` is an implementation of trait `simperby_network::AuthorizedNetwork`, and it is an actual network that `SimperBy` module will work on.
- Added several test schemes (they are abstract for now)
- Changed the types of already known peers
- Corrected typos

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I leave references, if exist, about this pull request (ex. discord message, google docs, etc).
- [ ] I have added thorough tests for core features.

## Checklist after creating a pull request
- [x] Mention reviews in the discord channel directly.

### CODE REVIEW

**Comment by lemonholic** [comment]
Removed `enum BootstrapPoint` and retrieved `known_peers` in the network inititation by commit [6de7ddb](https://github.com/postech-dao/simperby/pull/15/commits/6de7ddbf8a968445527f31ca64206ee4b568de33).

File: network/src/devnet.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    async fn get_live_list(&self) -> Result<Vec<PublicKey>, ()> {
+        unimplemented!("not implemented");
+    }
+}
+
+#[cfg(test)]
+mod test {
+    #[test]
+    #[ignore]
+    /// Test if this node receives a message from another node.
```
Comment (junha1) [inline network/src/devnet.rs:45]: 주석마다 `this node`라고 표현하는데, 테스트 코드 입장에서는 노드를 여러개 만들고 (대칭적인 상황이니까) 모든 노드에 대해서 같은 검사를 해보는거 아닌가요?

File: network/src/devnet.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        _known_peers: Vec<PublicKey>,
+        _bootstrap_points: Vec<SocketAddrV4>,
+        _network_id: String,
+    ) -> Result<Self, String>
+    where
+        Self: Sized,
+    {
+        unimplemented!("not implemented");
+    }
+    /// Broadcasts a message to the network, after signed by the key given to this instance.
```
Comment (junha1) [inline network/src/devnet.rs:27]: 특별히 코멘트할게 없으면 굳이 `AuthorizedNetwork`의 주석을 복붙할 이유는 없는 것 같습니다.

File: network/src/devnet.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
+use std::net::SocketAddrV4;
+
+use async_trait::async_trait;
+use tokio::sync::mpsc;
+
+use crate::AuthorizedNetwork;
+use simperby_common::crypto::*;
+
+/// An instance of `simperby::network::AuthorizedNetwork`
+pub struct DevNet {}
```
[the remark this answers is not in the document]
Comment (lemonholic) [reply network/src/devnet.rs:10]: I meant to introduce an actual backbone network where blocks and votes are propagated.
I'll consider changing its name to a more appropriate and intuitive one.

File: network/src/devnet.rs (hunk 2 above)
Comment (lemonholic) [reply network/src/devnet.rs:27]: Okay. I will remove the comments from the trait.

File: network/src/devnet.rs (hunk 1 above)
Comment (lemonholic) [reply network/src/devnet.rs:45]: OK. Surely all of the created nodes should be tested. I'll change the comments.

File: network/src/devnet.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (lemonholic) [reply network/src/devnet.rs:10]: How do you think of the name `network::PropagationNetwork`?

**Comment by lemonholic** [comment]
Applied all reviews above.

**Review by junha1 — APPROVED** [review]

**Pull request merged**